resources/user-agents/browsers/firefox/firefox-generic.json

```diff
@@ -26,77 +26,86 @@
           "match": "Mozilla/5.0 (#PLATFORM#) Gecko* Firefox/*",
           "device": "Linux Desktop",
           "platforms": [
-            "Ubuntu_64", "Ubuntu_12_04", "Ubuntu_32", "Ubuntu",
+            "Ubuntu_64", "Ubuntu_32", "Ubuntu",
             "Linux_32_on_x64", "Linux_64", "Linux_32", "Linux"
           ]
         },
         {
           "match": "Mozilla/5.0 (#PLATFORM#) Gecko* Firefox/*",
           "device": "Windows Desktop",
           "platforms": [
-            "NT4", "Win2000_64", "Win2000", "NT4_1",
+            "Win98", "Win95", "WinME",
+            "NT4", "NT4_B", "NT4_1",
+            "Win2000_64", "Win2000",
             "WinXPa_64", "WinXPa_32", "WinXPb_64", "WinXPb_32",
             "Vista_x64", "Vista_64", "Vista_32",
             "Win7_x64", "Win7_64", "Win7_32",
             "Win8_x64", "Win8_64", "Win8_32",
             "Win8_1_x64", "Win8_1_64", "Win8_1_32",
-            "Win10_x64", "Win10_64", "Win10_32",
-            "Win10_x64_B", "Win10_64_B", "Win10_32_B"
+            "Win10_x64", "Win10_64", "Win10_32", "Win10_x64_B", "Win10_64_B", "Win10_32_B"
           ]
         },
         {
           "match": "Mozilla/5.0 (#PLATFORM#) Gecko* Firefox/*",
           "device": "Macintosh",
           "platforms": [
-            "OSX_10_4", "OSX_10_5", "OSX_10_6", "OSX_10_7", "OSX_10_8", "OSX_10_9", "OSX_10_10",
-            "OSX_10_11",
+            "OSX_10_4", "OSX_10_5", "OSX_10_6", "OSX_10_7", "OSX_10_8", "OSX_10_9", "OSX_10_10", "OSX_10_11",
+            "OSX_10_12",
             "OSX",
-            "MacPPC_3"
+            "MacPPC"
           ]
         },
         {
           "match": "Mozilla/5.0 (#PLATFORM#) Gecko* Firefox/*",
           "device": "general Desktop",
           "platforms": [
-            "FreeBSD_64", "FreeBSD", "HPUX", "IRIX64", "OpenBSD_64_2", "OpenBSD", "SunOS", "OS2"
+            "FreeBSD_64", "FreeBSD",
+            "OpenBSD_64", "OpenBSD",
+            "NetBSD", "DragonFlyBSD",
+            "HPUX", "IRIX64", "BeOS", "SunOS", "OS2"
           ]
         },
         {
           "match": "Mozilla/4.0 (#PLATFORM#) Gecko* Firefox/*",
```

Comment:
I should mention that I'm suspicious of these `Mozilla/4.0` patterns, especially on newer versions of Firefox.  I don't think Firefox uses this prefix (not sure if it ever really did).

We do see a handful of useragents with this prefix that browscap has detected as Firefox, across a lot of different versions of Firefox (but at a much lower frequency than the `Mozilla/5.0` prefix) so I kept the patterns here (but didn't add any tests for it).

Perhaps re-visiting the Firefox source code to see if `Mozilla/4.0` could every happen legitimately is in order before making a decision on these.